From b9f1598eb38769a0801bb24d374d09c3aa575403 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jan-Simon=20M=C3=B6ller?= <dl9pf@gmx.de>
Date: Tue, 20 Nov 2012 21:18:16 +0100
Subject: [PATCH 08/39] crypto: LLVMLinux: aligned-attribute.patch

__attribute((aligned)) can only be used in type definitions, but
gcc allows it to be applied to a defined type.

This patch fixes errors when compiling the Linux kernel with Clang.
Breaking this code into 2 lines also makes it more understandable.

Author: Mark Charlebois <charlebm@gmail.com>
Signed-off-by: Mark Charlebois <charlebm@gmail.com>
---
 crypto/shash.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/crypto/shash.c b/crypto/shash.c
index 929058a..8fbb4c8 100644
--- a/crypto/shash.c
+++ b/crypto/shash.c
@@ -67,7 +67,8 @@ EXPORT_SYMBOL_GPL(crypto_shash_setkey);
 static inline unsigned int shash_align_buffer_size(unsigned len,
 						   unsigned long mask)
 {
-	return len + (mask & ~(__alignof__(u8 __attribute__ ((aligned))) - 1));
+	typedef __attribute__ ((aligned)) u8 u8_aligned;
+	return len + (mask & ~(__alignof__(u8_aligned) - 1));
 }
 
 static int shash_update_unaligned(struct shash_desc *desc, const u8 *data,
-- 
1.8.1.2

